pkg/sql/plan/prune.go

```diff
@@ -48,6 +48,12 @@ func (b *build) pruneExtend(e extend.Extend, isProjection bool) (extend.Extend,
 		if n.Op == overload.UnaryMinus {
 			return b.pruneUnaryMinus(n)
 		}
+	case *extend.MultiExtend:
```

Comment:
huh, I don't see the reason for this code

Reply:
The code here is to deal with the unary expression of - 1,  - 1 itself is a constant value. When calculating to the bottom layer, it should be trimmed to ExtendValue instead of UnaryeEtend. Otherwise, when substring (name, - 1), it will think that - 1 is an expression rather than a constant Value, but the judgment is wrong